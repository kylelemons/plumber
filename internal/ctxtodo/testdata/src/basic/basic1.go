// Copyright 2021 Kyle Lemons
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

package basic

import (
	"context"
	"log"
	"net"
)

func a() {
	ctx := context.TODO() // want "Plumb context"
	_ = ctx
}

func b(addr string) (net.Conn, error) {
	dialer := &net.Dialer{}
	return dialer.DialContext(context.TODO(), "tcp", addr) // want "Plumb context"
}

func c() {
	conn, err := b("localhost:12345")
	if err != nil {
		panic(err)
	}
	defer conn.Close()
	_ = context.TODO() // want "Plumb context"
}

func cycle1() {
	log.Println(context.TODO()) // want "Plumb context"
	cycle2()
}

func cycle2() {
	cycle1()
}

type t struct{}

func (t) m() { _ = context.TODO() } // want "Plumb context"
