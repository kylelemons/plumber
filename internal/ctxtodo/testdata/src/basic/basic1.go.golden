// Copyright 2021 Kyle Lemons
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

package basic

import (
	"context"
	"log"
	"net"
)

func a(ctx context.Context) {
	// want "Plumb context"
	_ = ctx
}

func b(ctx context.Context, addr string) (net.Conn, error) {
	dialer := &net.Dialer{}
	return dialer.DialContext(ctx, "tcp", addr) // want "Plumb context"
}

func c(ctx context.Context) {
	conn, err := b(ctx, "localhost:12345")
	if err != nil {
		panic(err)
	}
	defer conn.Close()
	// want "Plumb context"
}

func cycle1(ctx context.Context) {
	log.Println(ctx) // want "Plumb context"
	cycle2(ctx)
}

func cycle2(ctx context.Context) {
	cycle1(ctx)
}

type t struct{}

func (t) m(ctx context.Context) {} // want "Plumb context"
